pages/pokedex.tsx

```diff
@@ -21,54 +15,48 @@ function Pokedex() {
 
   const theme = `linear(to-l,#${firstValue},#${secondValue})`;
 
-  let offset = parseInt((router.query.offset as string) ?? "1");
-
-  const { isLoading, error, data } = useQuery(["content", offset], async () => {
-    const response = await axios.get(
+  const fetch = ({ pageParam: offset = 1 }) => {
+    return axios.get(
       `http://localhost:8081/api/pokemon?offset=${offset}&pageSize=24`
     );
-    const data = await response.data;
-    return data;
-  });
+  };
 
-  useEffect(() => {
-    if (data) {
-      setPokemon(data.content);
-      setPages(data.totalPages);
+  const { isLoading, error, data, fetchNextPage } = useInfiniteQuery(
+    ["content"],
+    fetch,
+    {
+      getNextPageParam: (lastPage, pages) => {
+        if (!lastPage.data.last) {
+          return lastPage.data.pageable.pageNumber + 2;
+        } else {
+          return undefined;
+        }
+      },
+      keepPreviousData: true,
     }
-  }, [offset, data]);
+  );
+
+  const pokemon = data?.pages.flatMap(({ data }) => data.content) ?? [];
 
   if (isLoading) return <Spinner />;
 
   if (error) return "An error has occured";
 
-  function forward() {
-    if (offset < pages) {
-      const currentPage = offset + 1;
-      router.push(
-        `/pokedex?offset=${currentPage}&firstValue=${firstValue}&secondValue=${secondValue}`
-      );
-    }
-  }
-
-  function back() {
-    if (offset > 1) {
-      const currentPage = offset - 1;
-      router.push(
-        `/pokedex?offset=${currentPage}&firstValue=${firstValue}&secondValue=${secondValue}`
-      );
-    }
+  function morePokemon() {
+    fetchNextPage();
+    console.log("clicked");
```

Comment:
We typically don't leave any console.log statements in our production code. If something needs visibility, we have other monitoring tools we wire up like [DataDog](https://datadog.bushelops.com/) so we can see the server errors generated, and also see client-side error reports through what is called [APM](https://app.datadoghq.com/apm/home) & [RUM](https://app.datadoghq.com/rum/performance-monitoring)